docs/spec_draft.md

```diff
@@ -1808,6 +1809,64 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.while
+
+### Semantics
+Produces the output from executing `body` function 0 or more times till the
+`cond` function outputs `true`. More formally, `body` function is executed once
+every time the `cond` function returns `true` and all execution stops once
+`cond` function returns `false`. For the first iteration both `cond` and `body`
+arguments are initialzed with the vaules of arguments of the `while` operation,
+for subsequent iterations they are fed with the output of `body` from the
+previous itertaion.
```

Comment:
Consider expressing the formalization in pseudocode - I think that'll be much easier to read. For example: 

"""
More formally, the semantics can be expressed as follows using Python-like 
syntax:

```C++
internal_state = operands;
while (cond(internal_state) == true) {
  internal_state = body(internal_state);
}
results = internal_state;
```
"""

Reply:
Thanks this looks good. Concise and expressive.